src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -10432,11 +10440,26 @@ public void FormName_NotAForm()
         AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
     }
 
+    [Fact, WorkItem("https://github.com/dotnet/razor/issues/9077")]
+    public void FormName_MissingUsing()
+    {
+        // Act
+        var generated = CompileToCSharp("""
+            <form method="post" @onsubmit="() => { }" @formname="named-form-handler"></form>
+            <form method="post" @onsubmit="() => { }" @formname="@("named-form-handler")"></form>
```

Comment:
For this test it appears no code is really generated and no diagnostics are emitted. Is that correct? If so it seems a bit odd to me that we do nothing and don't alert the user.

Reply:
The `@formname` is translated into `AddNamedEvent` normally - https://github.com/dotnet/razor/blob/f808c223fdbdbc2a05069005c9693022d4cc693e/src/Compiler/Microsoft.AspNetCore.Razor.Language/test/TestFiles/IntegrationTests/ComponentRuntimeCodeGenerationTest/FormName_MissingUsing/TestComponent.codegen.cs. The `@onsubmit` is emitted as-is but that's a pre-existing behavior - all events behave like this when the event handler is not registered.